nusantara/nusa_datasets/inset_lexicon/inset_lexicon.py

```diff
@@ -0,0 +1,126 @@
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from nusantara.utils import schemas
+from nusantara.utils.configs import NusantaraConfig
+from nusantara.utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{inproceedings,
+author = {Koto, Fajri and Rahmaningtyas, Gemala},
+year = {2017},
+month = {12},
+pages = {},
+title = {InSet Lexicon: Evaluation of a Word List for Indonesian Sentiment Analysis in Microblogs},
+doi = {10.1109/IALP.2017.8300625}
+}
+"""
+
+_DATASETNAME = "inset_lexicon"
+
+_DESCRIPTION = """\
+InSet, an Indonesian sentiment lexicon built to identify written opinion and categorize it into positive or negative opinion,
+which could be utilized to analyze public sentiment towards particular topic, event, or product. Composed using collection
+of words from Indonesian tweet, InSet was constructed by manually weighting each words and enhanced by adding stemming and synonym set
+"""
+
+_HOMEPAGE = "https://www.researchgate.net/publication/321757985_InSet_Lexicon_Evaluation_of_a_Word_List_for_Indonesian_Sentiment_Analysis_in_Microblogs"
+_LICENSE = "Unknown"
+_URLS = {_DATASETNAME: "https://github.com/fajri91/InSet/archive/refs/heads/master.zip"}
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
+_SOURCE_VERSION = "1.0.0"
+_NUSANTARA_VERSION = "1.0.0"
+
+
+class InsetLexicon(datasets.GeneratorBasedBuilder):
+    """InSet, an Indonesian sentiment lexicon built to identify written opinion and categorize it into positive or negative opinion"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    NUSANTARA_VERSION = datasets.Version(_NUSANTARA_VERSION)
+
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="inset_lexicon_source",
+            version=SOURCE_VERSION,
+            description="Inset Lexicon source schema",
+            schema="source",
+            subset_id="inset_lexicon",
+        ),
+        NusantaraConfig(
+            name="inset_lexicon_nusantara_text",
+            version=NUSANTARA_VERSION,
+            description="Inset Lexicon Nusantara schema",
+            schema="nusantara_text",
+            subset_id="inset_lexicon",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "inset_lexicon_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features({"word": datasets.Value("string"), "weight": datasets.Value("string")})
+        elif self.config.schema == "nusantara_text":
+            labels = list(range(-5, 6, 1))
+            labels = [str(label) for label in labels]
+            features = schemas.text_features(labels)
+
```

Comment:
Framing this problem as:
Word-level sentiment analysis and use the weight as label from -5 to 5

Reply:
Interesting, I think we can use this convention